### DISCUSSION TITLE
How can I open the Modelica.Math.Matrices.LAPACK library?

### DISCUSSION THREAD

**Discussion by Amuri44** [modelica/ModelicaStandardLibrary#3949] [Q&A]
Hello
I want to reach and see the contents of LAPACK library, but I could find it in the `OpenModelica1.18.0-64bit` folder?

**Comment by casella** [comment]
There are many external functions in Modelica.Math.Matrices.LAPACK that wrap LAPACK routines. You can call them like any other Modelica function.

**Reply by Amuri44** [reply]
Thanks for your reply!
Yes, so where can I find the library or source code of these functions in the installed openModelica files?

**Reply by beutlich** [reply]
Maybe https://doc.modelica.org/Modelica%204.0.0/Resources/helpOM/Modelica.Math.Matrices.LAPACK.html is of use here.

**Comment by casella** [comment]
I guess @AnHeuermann or @mahge know

**Reply by HansOlsson** [reply]
To me this discussion seems very focused on where they are in OpenModelica, and not generally applicable to MSL.

**Reply by casella** [reply]
> To me this discussion seems very focused on where they are in OpenModelica, and not generally applicable to MSL.

Yes, sorry, while answering I got confused about which issue tracker I was on 😄 

BTW, @Amuri44, I think the idea in general is that you should not call those functions directly, because tool vendors (any tool vendor, not only OSMC) may decide to legitimately move them around at any time, and update the tool setup so accordingly so that the external functions can still access them. So, if you use the external function interface, they will still work, while if you call them directly, they may break.

**Reply by sjoelund** [reply]
For Linux, it will not even be included in the OpenModelica install. It's a separate third-party library with many different implementations that can be interchanged (http://www.netlib.org/lapack/ would be the default, but you might be using ATLAS, OpenBLAS, MKL, etc). No header is necessary, only the library.

**Reply by Amuri44** [reply]
@sjoelund Thank you for your reply. So, I cannot find the library (as a file) in the installed folders of OpenModelica?

**Reply by sjoelund** [reply]
On Windows you can, but it's a binary object. No header or source code.

**Reply by Amuri44** [reply]
Nice, where I can find this binary object file (only for reference)?

**Reply by HansOlsson** [reply]
Can you kindly move this discussion to some more appropriate repository/forum since it does not seem to of general interest for MSL.
